src/components/ComparisonTable.tsx

```diff
@@ -0,0 +1,166 @@
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+import Paper from '@mui/material/Paper';
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+import DownloadsGraph from './DownloadsGraph';
+
+type Tcollected={
+  metadata: Tmetadata;
+  npm:Tnpm;
+}
+type Tevaluation={
+  popularity: {communityInterest:number;downloadsCount:number;};
+  quality:{carefulness: number; health: number,tests:number;};
+}
+type Tmetadata={
+  author:{name:string; email: string;};
+  description: string;
+  keywords: Array<string>;
+  links:{npm:string,homepage:string,respository: string};
+  maintainers: Array<Tmaintainers>;
+  publisher: {username: string, email: string};
+  name: string;
+  license:string;
+}
+
+type Tpackage={
+  analyzedAt: string;
+  collected: Tcollected;
+  evaluation: Tevaluation;
+}
+
+type Tnpm={
+  downloads: Tdownloads;
+  starsCount: number;
+}
+
+type Tdownloads={
+  index: number;
+  downloadsData:TdownloadsData; 
+}
+
+type TdownloadsData={
+  from: string;
+  to: string;
+  count: number;
+}
+
+type Tmaintainers={
+  username:string;
+   email:string;
+}
+
+type Trows={
+  packageName:string, firstPackageValue: string|number, secondPackageValue: string|number
+}
+
+
+export default function ComparisonTable({ selectedPackages }) {
+  const [firstPackagedata, setFirstPackageData] = useState<Tpackage| undefined>();
+  const [secondPackageData, setSecondPackageData] = useState<Tpackage| undefined>();
+  const [rows, setRows] = useState<Array<Trows>>([]);
+
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response1 = await axios.get<Tpackage>(`https://api.npms.io/v2/package/${selectedPackages[0].package.name}`);
+        setFirstPackageData(response1.data);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    fetchData();
+  }, [selectedPackages]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response2 = await axios.get<Tpackage>(`https://api.npms.io/v2/package/${selectedPackages[1].package.name}`);
+        setSecondPackageData(response2.data);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    fetchData();
+  }, [selectedPackages]);
+
+  useEffect(() => {
+    if (firstPackagedata&&secondPackageData) {
+      setRows([
+        createData('Description', firstPackagedata.collected.metadata.name, secondPackageData.collected.metadata.description),
```

Comment:
Move this formatting function to the utils file. Pass the function only the `firstPackagedData` and `secondPackageData` objects and handle the rest of the logic there. The return value from there should then be set in the state. 
This approach would make the code more readable and also separate the business logic.